Add examples of Apache Ignite 2.6.0 basic JMX metrics

Mmuzaf opened the pull request:
@brian-brazil , @skonto 

Hi Brian, Stavros 

Can you review my changes?
I've added basic JMX metrics for Apache Ignite [https://ignite.apache.org](https://ignite.apache.org)/ project for version 2.6.0. Hope they will help somebody to use this tool.

Kind regards,
Maxim Muzafarov

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
We don't put this in these files.

Mmuzaf replied:
Removed.

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+lowercaseOutputLabelNames: true
+lowercaseOutputName: true
+rules:
+# Apache Ignite provide runtime information on a cluster.  Keep in mind that there will be a certain
```
It'd be better to get the data from the individual processes directly, rather than adding jitter and lag.

Mmuzaf replied:
Brian, Could you, please, calrify? In general, I agree that it's better to share individual processes metrics. But I would like to have an batch of heap and non-heap metrics for the whole cluster.

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+  name: ignite_cluster_heap_$1
+  help: Ignite cluster amount of heap memory in bytes
+  labels:
+    attr: $3
```
Metric names should be in the name

Mmuzaf replied:
Agree. Added.

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+#
+# see https://ignite.apache.org/releases/latest/javadoc/org/apache/ignite/cluster/ClusterMetrics.html
+- pattern: "^org.apache<clsLdr=(.+), group=Kernal, name=ClusterMetrics(.+)><>(HeapMemory.*):"
+  name: ignite_cluster_heap_$1
```
clsLdr sounds like a label - does it vary as the leader changes?

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+# see https://ignite.apache.org/releases/latest/javadoc/org/apache/ignite/cluster/ClusterMetrics.html
+- pattern: "^org.apache<clsLdr=(.+), group=Kernal, name=ClusterMetrics(.+)><>(HeapMemory.*):"
+  name: ignite_cluster_heap_$1
+  help: Ignite cluster amount of heap memory in bytes
```
If you know it's in bytes, append _bytes to the name

Mmuzaf replied:
Agree. Added. JVM metrics used as an example.

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+  help: Ignite durable memory pages currently loaded in RAM
+  labels:
+    attr: $3
+- pattern: "^org.apache<clsLdr=(.+), group=DataRegionMetrics, name=(.+)><>(Off[Hh]eap.*|MaxSize|PhysicalMemorySize|TotalAllocatedSize):"
```
Are these the usual JVM memory metrics?

Mmuzaf replied:
Brian, no these metrics are collected through the whole cluster on specific `memory region`. They are not bounded to JVMs.

brian-brazil commented on example_configs/ignite_2.6.0.yml:
```diff
@@ -0,0 +1,95 @@
+  help: Ignite cluster job detalization
+  labels:
+    attr: $3
+  type: COUNTER
```
Counters should end in _total

Mmuzaf commented:
Example,

`# HELP ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate Ignite cache "test-cache" rebalancing rate
# TYPE ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate untyped
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate{attr="RebalancingKeysRate",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate{attr="RebalancingBytesRate",cache="\"test-cache\"",} 0.0
# HELP ignite_18b4aac2_dpl_mem_plc_pages_total Ignite durable memory pages currently loaded in RAM
# TYPE ignite_18b4aac2_dpl_mem_plc_pages_total gauge
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="DirtyPages",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="PhysicalMemoryPages",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="TotalAllocatedPages",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="AllocationRate",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="CheckpointBufferPages",} 0.0
# HELP ignite_18b4aac2_clustermetricsmxbeanimpl_job_total Ignite cluster job detalization
# TYPE ignite_18b4aac2_clustermetricsmxbeanimpl_job_total counter
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="TotalRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageActiveJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="TotalExecutedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentCancelledJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumActiveJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentActiveJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageWaitingJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumCancelledJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="TotalCancelledJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumWaitingJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentWaitingJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageCancelledJobs",} 0.0
# HELP ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes Ignite cluster amount of heap memory in bytes
# TYPE ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes gauge
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryMaximum",} 1.053294592E9
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryInitialized",} 1.073741824E9
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryCommitted",} 1.053294592E9
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryUsed",} 1.46684384E8
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryTotal",} 1.053294592E9
# HELP ignite_18b4aac2_dpl_mem_plc_pages_bytes Ignite durable memory pages size metrics in bytes
# TYPE ignite_18b4aac2_dpl_mem_plc_pages_bytes gauge
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="MaxSize",} 600.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="OffheapUsedSize",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="OffHeapSize",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="TotalAllocatedSize",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="PhysicalMemorySize",} 0.0
# HELP ignite_18b4aac2_datastoragemetrics_pds_wal Ignite WAL metrics
# TYPE ignite_18b4aac2_datastoragemetrics_pds_wal untyped
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 6.7108864E7
# HELP ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left Ignite cache "test-cache" rebalancing data left
# TYPE ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left counter
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left{attr="RebalanceClearingPartitionsLeft",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left{attr="KeysToRebalanceLeft",cache="\"test-cache\"",} 0.0
# HELP ignite_18b4aac2_metastorememplc_pages_total Ignite durable memory pages currently loaded in RAM
# TYPE ignite_18b4aac2_metastorememplc_pages_total gauge
ignite_18b4aac2_metastorememplc_pages_total{attr="DirtyPages",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="PhysicalMemoryPages",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="TotalAllocatedPages",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="AllocationRate",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="CheckpointBufferPages",} 0.0
# HELP ignite_18b4aac2_sysmemplc_pages_bytes Ignite durable memory pages size metrics in bytes
# TYPE ignite_18b4aac2_sysmemplc_pages_bytes gauge
ignite_18b4aac2_sysmemplc_pages_bytes{attr="MaxSize",} 100.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="OffheapUsedSize",} 0.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="OffHeapSize",} 0.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="TotalAllocatedSize",} 0.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="PhysicalMemorySize",} 0.0
# HELP ignite_18b4aac2_sysmemplc_pages_total Ignite durable memory pages currently loaded in RAM
# TYPE ignite_18b4aac2_sysmemplc_pages_total gauge
ignite_18b4aac2_sysmemplc_pages_total{attr="DirtyPages",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="PhysicalMemoryPages",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="TotalAllocatedPages",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="AllocationRate",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="CheckpointBufferPages",} 0.0
# HELP ignite_18b4aac2_metastorememplc_pages_bytes Ignite durable memory pages size metrics in bytes
# TYPE ignite_18b4aac2_metastorememplc_pages_bytes gauge
ignite_18b4aac2_metastorememplc_pages_bytes{attr="MaxSize",} 100.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="OffheapUsedSize",} 0.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="OffHeapSize",} 0.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="TotalAllocatedSize",} 0.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="PhysicalMemorySize",} 0.0
# HELP ignite_18b4aac2_datastoragemetrics_pds_checkpoint Ignite persistence last checkpoint metrics
# TYPE ignite_18b4aac2_datastoragemetrics_pds_checkpoint counter
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
# HELP ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes Ignite cluster amount of non-heap memory in bytes
# TYPE ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes gauge
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryInitialized",} 2555904.0
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryUsed",} 6.5789616E7
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryCommitted",} 6.8313088E7
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryMaximum",} 1.59383552E9
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryTotal",} 1.59383552E9
# HELP ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations Ignite cache "test-cache" operations metrics
# TYPE ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations counter
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheGets",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CachePuts",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheMisses",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheRemovals",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheHits",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheEvictions",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheMissPercentage",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheHitPercentage",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheTxRollbacks",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheTxCommits",cache="\"test-cache\"",} 0.0
`

Mmuzaf commented:
Example,

```
# HELP ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate Ignite cache "test-cache" rebalancing rate
# TYPE ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate untyped
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate{attr="RebalancingKeysRate",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_rate{attr="RebalancingBytesRate",cache="\"test-cache\"",} 0.0
# HELP ignite_18b4aac2_dpl_mem_plc_pages_total Ignite durable memory pages currently loaded in RAM
# TYPE ignite_18b4aac2_dpl_mem_plc_pages_total gauge
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="DirtyPages",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="PhysicalMemoryPages",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="TotalAllocatedPages",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="AllocationRate",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_total{attr="CheckpointBufferPages",} 0.0
# HELP ignite_18b4aac2_clustermetricsmxbeanimpl_job_total Ignite cluster job detalization
# TYPE ignite_18b4aac2_clustermetricsmxbeanimpl_job_total counter
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="TotalRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageActiveJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="TotalExecutedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentCancelledJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumActiveJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentActiveJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageWaitingJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumCancelledJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageRejectedJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="TotalCancelledJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="MaximumWaitingJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="CurrentWaitingJobs",} 0.0
ignite_18b4aac2_clustermetricsmxbeanimpl_job_total{attr="AverageCancelledJobs",} 0.0
# HELP ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes Ignite cluster amount of heap memory in bytes
# TYPE ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes gauge
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryMaximum",} 1.053294592E9
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryInitialized",} 1.073741824E9
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryCommitted",} 1.053294592E9
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryUsed",} 1.46684384E8
ignite_18b4aac2_clustermetricsmxbeanimpl_heap_bytes{attr="HeapMemoryTotal",} 1.053294592E9
# HELP ignite_18b4aac2_dpl_mem_plc_pages_bytes Ignite durable memory pages size metrics in bytes
# TYPE ignite_18b4aac2_dpl_mem_plc_pages_bytes gauge
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="MaxSize",} 600.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="OffheapUsedSize",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="OffHeapSize",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="TotalAllocatedSize",} 0.0
ignite_18b4aac2_dpl_mem_plc_pages_bytes{attr="PhysicalMemorySize",} 0.0
# HELP ignite_18b4aac2_datastoragemetrics_pds_wal Ignite WAL metrics
# TYPE ignite_18b4aac2_datastoragemetrics_pds_wal untyped
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_wal{attr="DataStorageMetrics",} 6.7108864E7
# HELP ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left Ignite cache "test-cache" rebalancing data left
# TYPE ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left counter
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left{attr="RebalanceClearingPartitionsLeft",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_rebalance_left{attr="KeysToRebalanceLeft",cache="\"test-cache\"",} 0.0
# HELP ignite_18b4aac2_metastorememplc_pages_total Ignite durable memory pages currently loaded in RAM
# TYPE ignite_18b4aac2_metastorememplc_pages_total gauge
ignite_18b4aac2_metastorememplc_pages_total{attr="DirtyPages",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="PhysicalMemoryPages",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="TotalAllocatedPages",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="AllocationRate",} 0.0
ignite_18b4aac2_metastorememplc_pages_total{attr="CheckpointBufferPages",} 0.0
# HELP ignite_18b4aac2_sysmemplc_pages_bytes Ignite durable memory pages size metrics in bytes
# TYPE ignite_18b4aac2_sysmemplc_pages_bytes gauge
ignite_18b4aac2_sysmemplc_pages_bytes{attr="MaxSize",} 100.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="OffheapUsedSize",} 0.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="OffHeapSize",} 0.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="TotalAllocatedSize",} 0.0
ignite_18b4aac2_sysmemplc_pages_bytes{attr="PhysicalMemorySize",} 0.0
# HELP ignite_18b4aac2_sysmemplc_pages_total Ignite durable memory pages currently loaded in RAM
# TYPE ignite_18b4aac2_sysmemplc_pages_total gauge
ignite_18b4aac2_sysmemplc_pages_total{attr="DirtyPages",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="PhysicalMemoryPages",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="TotalAllocatedPages",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="AllocationRate",} 0.0
ignite_18b4aac2_sysmemplc_pages_total{attr="CheckpointBufferPages",} 0.0
# HELP ignite_18b4aac2_metastorememplc_pages_bytes Ignite durable memory pages size metrics in bytes
# TYPE ignite_18b4aac2_metastorememplc_pages_bytes gauge
ignite_18b4aac2_metastorememplc_pages_bytes{attr="MaxSize",} 100.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="OffheapUsedSize",} 0.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="OffHeapSize",} 0.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="TotalAllocatedSize",} 0.0
ignite_18b4aac2_metastorememplc_pages_bytes{attr="PhysicalMemorySize",} 0.0
# HELP ignite_18b4aac2_datastoragemetrics_pds_checkpoint Ignite persistence last checkpoint metrics
# TYPE ignite_18b4aac2_datastoragemetrics_pds_checkpoint counter
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
ignite_18b4aac2_datastoragemetrics_pds_checkpoint{attr="DataStorageMetrics",} 0.0
# HELP ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes Ignite cluster amount of non-heap memory in bytes
# TYPE ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes gauge
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryInitialized",} 2555904.0
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryUsed",} 6.5789616E7
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryCommitted",} 6.8313088E7
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryMaximum",} 1.59383552E9
ignite_18b4aac2_clustermetricsmxbeanimpl_nonheap_bytes{attr="NonHeapMemoryTotal",} 1.59383552E9
# HELP ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations Ignite cache "test-cache" operations metrics
# TYPE ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations counter
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheGets",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CachePuts",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheMisses",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheRemovals",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheHits",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheEvictions",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheMissPercentage",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheHitPercentage",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheTxRollbacks",cache="\"test-cache\"",} 0.0
ignite_18b4aac2_cacheclustermetricsmxbeanimpl_test_cache_operations{attr="CacheTxCommits",cache="\"test-cache\"",} 0.0
```

Mmuzaf commented:
@brian-brazil 
I've updated metrics according to [Metric and label naming](https://prometheus.io/docs/practices/naming/) and [Instrumentation](https://prometheus.io/docs/practices/instrumentation/) articles.  Also, I've updated the metric description (some of them disabled by default).

brian-brazil commented:
clsLdr is still in there, did you forget to push?

Mmuzaf commented:
@brian-brazil yes, I'll update PR shortly.

dhoard commented:
Closed due to inactivity.
